src/sourcerer/infrastructure/storage_provider/services/gcp.py

```
@@ -0,0 +1,261 @@
+        Raises:
+            UploadStorageItemsException: If an error occurs while uploading the item
+        """
+        try:
+            bucket = self.client.bucket(storage)
+            storage_path = str(Path(storage_path or "") / (dest_path or source_path.name))
+            bucket.blob(storage_path).upload_from_filename(source_path)
+        except Exception as ex:
+            raise UploadStorageItemsException(str(ex)) from ex
+
+    def download_storage_item(
+            self, storage: str, key: str, progress_callback: Callable | None = None
```

Ruff (0.8.2)

Unused method argument: `progress_callback`

(ARG002)